pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionConsumeTest.java

```diff
@@ -368,4 +373,44 @@ public void completed(Exception e, long ledgerId, long entryId) {
         return positionList;
     }
 
+    @Test
+    public void testAckChunkMessage() throws Exception {
```

Comment:
I reverted the changes in `ConsumerImpl` and this test still passed. Could you explain why?

Reply:
@liangyepianzhou I think you should change the subscription type to Shared or checking the backlog, because the unacked messages for Exclusive or Failover will always be 0.